app/models/gsa18f/procurement.rb

```diff
@@ -95,5 +99,11 @@ def self.user_with_role(role_name)
 
       users.first
     end
+
+    private
+
+    def purchase_step
+      steps.select{|step| step.is_a?(Steps::Purchase)}.first
```

Comment:
Because `find_by` will execute SQL every time, while `select` just operates on the array of related `Step` objects. So there is a single SQL statement here:

```ruby
if purchase_step
  purchase_step.user
end
```

rather than 2. Likewise, the next time you call `purchaser` it will execute zero SQL, rather than 2 more statements.